timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/LeaderLearnerResource.java

```diff
@@ -48,22 +50,37 @@ public LeaderLearnerResource(PaxosLearner learner) {
     @Path("learn/{seq:.+}")
     @Consumes(MediaType.APPLICATION_JSON)
     @Timed
-    public void learn(@PathParam("seq") long seq, PaxosValue val) {
+    @Handle(
+            method = HttpMethod.POST,
+            path = "/" + PaxosTimeLockConstants.INTERNAL_NAMESPACE
+                    + "/" + PaxosTimeLockConstants.LEADER_PAXOS_NAMESPACE
+                    + "/learner/learn/{seq}")
+    public void learn(@PathParam("seq") @Handle.PathParam long seq, @Handle.Body PaxosValue val) {
```

Comment:
In general, we are a bit inconsistent in the repo w.r.t. adding safety annotations here. I will add a Safe for the sequence number unless someone dissents internally or has more context - it should just be internal Paxos things.